### PR TITLE
Perf : No find entities used when defense are built are far from nest

### PR DESCRIPTION
**Pull request by Ragnarok77-factorio**
### Brief description of the changes:
Basically to improve performance : saw in a profiler than find entities is quite heavy when defense are being replaced automatically by bots, The idea was not to do this kind of find entities on 70 radius when it's far from nests.

Function was copied (with noise removed) because I couldn't import terrain from functions (too many c level..).
I did +140 distance to be sure that it's not nearby nests (to avoid any kind of exploit with this change, and anyways even if it's a bit a big number, since barely no ones build defense and have active defenses there, it should be fine, if you want we can reduce the 140 number, just arbitrary but ugh, felt correct enough when testing locally.

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Review by AwesomePatrol — CHANGES_REQUESTED** [review]
I would prefer the function to be the same across the implementations, so remove the one in `terrain.lua` and use the copied version from `functions.lua` there.

**Comment by Ragnarok77-factorio** [comment]
Okay, work in progress but for the note, we can't remove the noise, it becomes unnatural otherwise, will cut it in two, without noise in the method for biter area : 
image

**Comment by Ragnarok77-factorio** [comment]
> I would prefer the function to be the same across the implementations, so remove the one in `terrain.lua` and use the copied version from `functions.lua` there (`functions.lua` is already imported in `terrain.lua`). It doesn't matter if a few nests can be destroyed with a turret creep. We want to prevent easily clearing huge parts of nest area.

Understood, updated as you asked, moved what was needed to Functions and made them public. So no more duplicate methods.

File: maps/biter_battles_v2/functions.lua [hunk 1]
```diff
@@ -263,10 +265,60 @@ function Public.init_player(player)
...
 function Public.no_turret_creep(event)
 	local entity = event.created_entity
 	if not entity.valid then return end
 	if not no_turret_blacklist[event.created_entity.type] then return end
+	
+	local posEntity = entity.position
+	if posEntity.y > 0 then posEntity.y = (posEntity.y + 100) * -1 end
+	if posEntity.y < 0 then posEntity.y = posEntity.y - 100 end
+	if not Public.is_biter_area(posEntity,false) then
+		game.print('NOPE too far' .. entity.position.y)
```
Comment (AwesomePatrol) [inline maps/biter_battles_v2/functions.lua]: Just remember to remove this debug print before merge

**Review by AwesomePatrol — APPROVED** [review]

**Pull request merged**